x-pack/plugins/security_solution/public/common/components/exceptions/builder/helpers.tsx

```diff
@@ -282,6 +285,7 @@ export const getUpdatedEntriesOnDelete = (
       const { field } = itemOfInterest;
       const updatedItemOfInterest: EntryNested | EmptyNestedEntry = {
         field,
+        id: itemOfInterest.id ?? `${entryIndex}`,
```

Comment:
Will this exceptional case ever happen?

Reply:
`id` is mapped to `string | undefined` as I can't truly guarantee that the id will be there given it's not actually part of the schema so I hesitated to not use `id?`. The entries that are typed as empty entries have `id` set as just `string` (as opposed to `string | undefined`) because we're creating those client side and can guarantee we'll stick an `id` in there.